packages/core-assortments/db/helpers/build-paths.js

```diff
@@ -0,0 +1,56 @@
+const walkAssortmentLinks = resolveAssortmentLink => async rootAssortmentId => {
+  const walk = async (assortmentId, initialPaths = [], childAssortmentId) => {
+    const assortmentLink = await resolveAssortmentLink(
+      assortmentId,
+      childAssortmentId
+    );
+    if (!assortmentLink) return initialPaths;
+
+    const subAsssortmentLinks = await Promise.all(
+      assortmentLink.parentIds.map(async parentAssortmentId => {
+        return walk(parentAssortmentId, initialPaths, assortmentId);
+      })
+    );
+
+    if (subAsssortmentLinks.length > 0) {
+      return subAsssortmentLinks.map(subAsssortmentLink => {
+        return [...subAsssortmentLink.flat(), assortmentLink, ...initialPaths];
+      });
```

Comment:
you could do the mapping directly in subAsssortmentLinks, since you map twice over same-sized arrays

Reply:
I've been looking at that code since 10 minutes and i can't follow 🤪